ios/FluentUI/Navigation/UIBarButtonItem+RedDot.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc public extension UIBarButtonItem {
```

Comment:
We generally do not want to create public extensions to system classes in Fluent, as it forces all downstream consumers to deal with our extension in their namespace. Instead, either use a custom `UIBarButtonItem` subclass, or provide some form of helper class that can perform the mutation for you.

Reply:
But we are following same styling for Badge value as well. I am just adding one extra functionality on top of badge button by using same mechanism.

Reply:
The existence of technical debt does not automatically justify the addition of more. We should move the badge styling away from this pattern as well; adding more that needs to be changed will only increase future work, *especially* when it comes to public APIs that downstream partners expect to remain supported.